filament/backend/src/vulkan/VulkanPipelineCache.cpp

```diff
@@ -35,7 +35,7 @@ namespace filament::backend {
 
 static VulkanPipelineCache::RasterState createDefaultRasterState();
 
-static VkShaderStageFlags getShaderStageFlags(utils::bitset64 key, uint16_t binding) {
+static VkShaderStageFlags getShaderStageFlags(VulkanPipelineCache::UsageFlags key, uint16_t binding) {
     VkShaderStageFlags flags = 0;
     if (key.test(binding * 2 + 0)) {
         flags |= VK_SHADER_STAGE_VERTEX_BIT;
```

Comment:
this code makes me nervous for when we'll need to add a COMPUTE stage. Is there a way to make this less easy to miss?

Reply:
Let me add an assert in `setUsageFlags` (soon to be renamed) if any flags get passed in that are not VERTEX or FRAGMENT.  I will also add an assert that looks at PIPELINE_STAGE_COUNT, near the definition of PipelineLayoutKey, in case we need to change the bitset64 into something else.

Reply:
Ah, I can make this code less brittle if I used MAX_FRAGMENT_SAMPLER_COUNT and MAX_FRAGMENT_SAMPLER_COUNT.